### PR TITLE
[SPARK-35785][SS][FOLLOWUP] Remove ignored test from RocksDBSuite

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This patch removes an ignored test from `RocksDBSuite`.

### Why are the changes needed?

The removed test is now ignored. The test itself doesn't look making sense. For example, the condition for capturing exception is never matched. The test runs updates to RocksDB instances at same remote dir with same versions. This doesn't look like a case it will run through in practice.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by viirya** [comment]
cc @xuanyuanking If we don't need the test, I think we can just remove it.

**Comment by viirya** [comment]
cc @HeartSaVioR

**Review by dongjoon-hyun — COMMENTED** [review]
Hi, @viirya (cc @xuanyuanking, @HeartSaVioR ). In the PR description, please write the reason why it looks not making sense.
> We have discussed about the test offline and it looks not making sense.

In addition, you may want to remove the sentence `We have ... sense` completely. It looks like it means @viirya , @xuanyuanking , and @HeartSaVioR , but it's not clear who is **we** .

**Comment by viirya** [comment]
@dongjoon-hyun Yea, thanks. Actually I should move the "we ..." out of the description.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala [hunk 1]
```diff
@@ -337,69 +337,6 @@ class RocksDBSuite extends SparkFunSuite {
...
-                  version = version) { db =>
-                  val prevValue = Option(toStr(db.get("a"))).getOrElse("0").toInt
-                  db.put("a", (prevValue + 1).toString)
-                  db.commit()
-                }
-              }
-            } catch {
-              case e: Exception =>
-                val newException = new Exception(s"ThreadId ${this.getId} failed", e)
-                if (exception != null) {
```
Comment (viirya) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala:362]: First all, `exception` is null at beginning. So this condition will be never satisfied.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. At least, `if (exception != null) {` seems to be invalid.

**Comment by dongjoon-hyun** [comment]
Jenkins passed. Merged to master/3.2.

**Comment by viirya** [comment]
Thanks @dongjoon-hyun

**Comment by xuanyuanking** [comment]
Agree to delete this first.
Refer to https://github.com/apache/spark/pull/33220, even with the exception var fix, you may still found the test is flaky in Jenkins env.

**Comment by HeartSaVioR** [comment]
Thanks for dealing with this, @viirya !

**Pull request closed without merging**